src/groupeddataframe/groupeddataframe.jl

```diff
@@ -58,15 +58,21 @@ into row groups.
 # Arguments
 - `df` : an `AbstractDataFrame` to split
 - `cols` : data frame columns to group by. Can be any column selector
-  ($COLUMNINDEX_STR; $MULTICOLUMNINDEX_STR).
-- `sort` : if `sort=true` sort groups according to the values of the grouping columns
-  `cols`; if `sort=false` groups are created in their order of appearance in `df`
-  if `sort=nothing` (the default) then the fastest available grouping algorithm
-  is picked and in consequence the order of groups in the result is undefined
-  and may change in future releases; below a description of the current
-  implementation is provided. Additionally `sort` can be a `NamedTuple` having
-  some or all of `alg`, `lt`, `by`, `rev`, and `order` fields. In this case
-  the groups are sorted and their order follows the [`sortperm`](@ref) order.
+  ($COLUMNINDEX_STR; $MULTICOLUMNINDEX_STR). In particular if the selector
+  picks no columns then a single-group `GroupedDataFrame` is created. As a
+  special case, if a list of columns to group by is passed as a vector it can
+  contain columns wrapped in [`order`](@ref) that will be used to determine
+  the order of groups if `sort` is `true` or a `NamedTuple` (if `sort` is
+  `nothing` or `false`, then passing `order` is an error).
```

Comment:
Thinking about it again, maybe it would be logical and convenient to enable sorting when `sort=nothing` and and `order` wrapper is passed? Can also be done later as it wouldn't be breaking.

Reply:
I have changed it the way you proposed. Indeed it will be more convenient.